### PR TITLE
Task 1

### PR DESCRIPTION
**Pull request by lopharb**

### CODE REVIEW

**Comment by foxxyanya** [comment]
Please check our branch naming convention in Miro 
`task1` -> `DS-1.1`

File: notebooks/dqc.ipynb [hunk 1]
```diff
@@ -0,0 +1,2357 @@
...
+   "metadata": {},
+   "source": [
+    "found no `NaN`s, but found an item with negative price, which has to be removed later"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Looking for unresolved relations between tables\n",
```
Comment (foxxyanya) [inline notebooks/dqc.ipynb:560]: Another point that has to be checked here is whether there are any shops/items, that are represented in test set, but are missing in training set.

File: notebooks/dqc.ipynb [hunk 2]
```diff
@@ -0,0 +1,2357 @@
...
+    "# items with unknown category\n",
+    "\n",
+    "items[items.merge(item_categories, on='item_category_id', how='left')['item_category_name'].isna()]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### looking for noisy text values\n",
```
Comment (foxxyanya) [inline notebooks/dqc.ipynb:845]: Plz add a summary: what shops/items are themselves identical but are represented with different names and how do we map them in the preprocessing stage?

E.g. as for "Жуковский ул. Чкалова 39м2" - are there any other examples?

File: notebooks/dqc.ipynb [hunk 3]
```diff
@@ -0,0 +1,2357 @@
...
+    "        plt.subplot(1, len(column_names), index+1)\n",
+    "        sns.boxplot(df[col])\n",
+    "    plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## plotting the data"
```
Comment (foxxyanya) [inline notebooks/dqc.ipynb:1319]: In case of `item_price`, `item_cnt_day`, the distribution seems to be near-exponential. Why don't we use log scale for y axis, so that the hist plots you're drawing are more informative?

File: notebooks/dqc.ipynb [hunk 4]
```diff
@@ -0,0 +1,2357 @@
...
+   ],
+   "source": [
+    "items.merge(sales['item_id'].value_counts(), on='item_id', how='inner').sort_values('count', ascending=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## looking for a way to remove outliers\n",
```
Comment (foxxyanya) [inline notebooks/dqc.ipynb:2077]: Actually the z-score idea may be good in general, but it's out of scope for DQC for the next reason. 

The goal of outliers check in DQC is **not** to remove outliers _in statistical sense_, instread, we strive to remove them in more straightforward way. 

There's no way actually to create a universal outlier removal mechanism for all the models you'll test in the future. Some models are more sensitive to outliers (e.g. linear models), some of them are not that sensitive. So for this stage, we just want to remove outliers that don't make any sense in terms of business logic.

For example, you current approach may  remove usual shopping bags, because  their amount sold is large enough. But that's not obviously an outlier in terms of business logic, and in the future you may want to predict sales for shopping bags as well.

Comment (foxxyanya) [reply notebooks/dqc.ipynb:2077]: I just suggest to move the logic with z-scores to EDA, and check here if there're any inconsistencies in terms of business logic (both for count and price features)

**Review by foxxyanya — CHANGES_REQUESTED** [review]
Overall looks good. Let's just make small changes that I noted to DQC (and reflect them in ETL script) and we'll move forward to EDA and modeling

File: notebooks/dqc.ipynb (hunk 1 above)
Comment (lopharb) [reply notebooks/dqc.ipynb:560]: I think I did it a few cells below by left joining the `test` to `shops` and `items` respectively and then seeing if there are any `NaN`s (which would appear in case of values that are present only in `test`).
